oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/ebpf/analyze/EBPFProfilingAnalyzer.java

```diff
@@ -0,0 +1,157 @@
+
+public class EBPFProfilingAnalyzer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EBPFProfilingAnalyzer.class);
+
+    private static final EBPFProfilingAnalyzeCollector ANALYZE_COLLECTOR = new EBPFProfilingAnalyzeCollector();
+    private static final Long FETCH_FETCH_DURATION = TimeUnit.MINUTES.toMillis(2);
+
+    private final ModuleManager moduleManager;
+    protected IEBPFProfilingDataDAO dataDAO;
+
+    public EBPFProfilingAnalyzer(ModuleManager moduleManager) {
+        this.moduleManager = moduleManager;
+    }
+
+    /**
+     * search data and analyze
+     */
+    public EBPFProfilingAnalyzation analyze(String taskId, List<EBPFProfilingAnalyzeTimeRange> ranges) throws IOException {
+        EBPFProfilingAnalyzation analyzation = new EBPFProfilingAnalyzation();
+
+        List<TimeRange> timeRanges = buildTimeRanges(ranges);
+        if (CollectionUtils.isEmpty(timeRanges)) {
+            analyzation.setTip("data not found");
+            return analyzation;
+        }
+
+        // query data
+        List<EBPFProfilingStack> stacks = timeRanges.parallelStream().map(r -> {
+            try {
+                return getDataDAO().queryData(taskId, r.minTime, r.maxTime);
+            } catch (IOException e) {
+                LOGGER.warn(e.getMessage(), e);
+                return Collections.<EBPFProfilingDataRecord>emptyList();
+            }
+        }).flatMap(Collection::stream).map(e -> {
+            try {
+                return EBPFProfilingStack.deserialize(e);
+            } catch (Exception ex) {
+                LOGGER.warn("could not deserialize the stack", ex);
+                return null;
+            }
+        }).filter(Objects::nonNull).distinct().collect(Collectors.toList());
+
+        // analyze
+        final List<EBPFProfilingTree> trees = analyze(stacks);
+        if (trees != null) {
+            analyzation.getTrees().addAll(trees);
+        }
+
+        return analyzation;
+    }
+
+    protected List<TimeRange> buildTimeRanges(List<EBPFProfilingAnalyzeTimeRange> timeRanges) {
+        return timeRanges.parallelStream()
+                .map(r -> buildTimeRanges(r.getStart(), r.getEnd()))
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    protected List<TimeRange> buildTimeRanges(long start, long end) {
+        if (start >= end) {
+            return null;
+        }
+
+        // include latest millisecond
+        end += 1;
+
+        final List<TimeRange> timeRanges = new ArrayList<>();
+        do {
+            long batchEnd = Math.min(start + FETCH_FETCH_DURATION, end);
+            timeRanges.add(new TimeRange(start, batchEnd));
+            start = batchEnd;
+        }
+        while (start < end);
+
+        return timeRanges;
+    }
+
+    /**
+     * Analyze records
+     */
+    protected List<EBPFProfilingTree> analyze(List<EBPFProfilingStack> stacks) {
```

Comment:
This method should have very strong UTs to cover merging logic. This is the key of the whole visualization result.

Reply:
Also, we need some benchmarks for this merging, as you provide a list of durations, so we could face a larger merging.

At the same time, we need a `circuit break` to avoid timeout. We should have the capability to provide a half-finished profiling tree with a message of `circuit break`.

Reply:
I have added a [benchmark](https://github.com/apache/skywalking/pull/8696/files#diff-15d820525f50abd439c83f34d6afd86a01134ed964142dc0fce907b7c1d4cc33) working on checking the analysis time without query data. I think profiling data analysis is not a big problem, we should add the `circuit break` when the query timeout. WDYT?

Reply:
Make sense.

Reply:
Add a new configuration item for the query timeout. default `30s`.